### PR TITLE
Feature/Aggregate Fns 

### LINKED ISSUE #64: Add Filter2D infr to lib_nn
Add all associated Classes and Tests to support Filter2D kernels

### PR DESCRIPTION
**Pull request by andrewstanfordjason**

### CODE REVIEW

**Review by ghost — APPROVED** [review]
+1

File: lib_nn/api/AggregateFn.hpp [hunk 1]
```diff
@@ -0,0 +1,578 @@
...
+#include "xs3_vpu.h"
+
+namespace nn {
+
+class AggregateFn {
+ public:
+  /**
+   * @brief Function to aggregate an input receptive field as defined by the
+   * operation it is performing, i.e. convolution, maximum, addition, etc.
+   * Arrgreation is preformed in batches of channel groups, each of the same
```
Comment (alexandracraciun) [inline lib_nn/api/AggregateFn.hpp]: Check syntax in class AggregateFn oxygen comment.

Comment (andrewstanfordjason) [reply lib_nn/api/AggregateFn.hpp]: what do you mean?

Comment (alexandracraciun) [reply lib_nn/api/AggregateFn.hpp]: * Aggregation is performed in batches of

**Pull request merged**